pydantic/_internal/_fields.py

```diff
@@ -248,7 +248,11 @@ def collect_dataclass_fields(
         if is_classvar(ann_type):
             continue
 
-        if not dataclass_field.init and dataclass_field.default_factory == dataclasses.MISSING:
+        if (
+            not dataclass_field.init
+            and dataclass_field.default_factory == dataclasses.MISSING
+            and dataclass_field.default == dataclasses.MISSING
```

Comment:
I would have swapped the order because `default` is used more often that `default_factory`

Reply:
Sure! Done